frontend/src/lib/ForwardMessageCache.test.ts

```diff
@@ -119,17 +121,40 @@ test("caches messages correctly", async () => {
     msg3.metadata.deltaPath = [2]
   }
   const ref = createRefMsg(msg3)
-  const unreferenced = await cache.processMessagePayload(ref)
+  const encodedRefMsg = ForwardMsg.encode(ref).finish()
+  const unreferenced = await cache.processMessagePayload(ref, encodedRefMsg)
   expect(getCachedMessage(ref.hash)).toBeUndefined()
   expect(unreferenced).toEqual(msg3)
 
   // Test that our uncached messages are copies
   expect(unreferenced).not.toBe(msg3)
 })
 
+test("caches messages as a deep copy", async () => {
+  const { cache, getCachedMessage } = createCache()
+
+  const msg = createForwardMsg("Cacheable", true)
+  const encodedMsg = ForwardMsg.encode(msg).finish()
+
+  await cache.processMessagePayload(msg, encodedMsg)
+
+  // Check if message is correctly cached
+  expect(getCachedMessage("Cacheable")).toEqual(msg)
+  expect(getCachedMessage("Cacheable")?.metadata?.deltaPath).toEqual([])
+
+  // Modify the message content
+  if (msg.metadata) {
+    msg.metadata.deltaPath = [10]
+  }
```

Comment:
I don't think we want to conditionally modify the message in this test (what if `msg.metadata` is null? is the test skipped). Better to unconditionally construct a new msg.metadata object with the exact value we care about.

Reply:
I changed it to `msg.metadata!.deltaPath = [10]`. This only modifies the value inside the object structure and throws an exception if it is null, so the test should work as expected.